scalableminds/webknossos
# Merge Volume Tracings

### TODO
 - [x] implement mergeFromIds route
 - [x] implement mergedFromContents route, send volume zips in initialData (in a tar? or multipart-request?)
 - [x] implement deterministic ordering (last modified annotation wins)
 - [x] save buckets to temporary store when appropriate, handle this when loading buckets
 - [x] send hard errors when loading buckets for tracing that is removed from temporary store, encouraging user to reload. Q: where to get this information? [store in redis.] How long should the cache timeout be? [1h] Can we re-merge in the backend if tracing is still active? [no, let user reload]
 - [x] relabel to avoid merging cells?
 - [x] copy to my account for temporary volume data?
 - [x] test for volume tracings with 8/16/32-bit segmentation fallback layer

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- upload zipfile with multiple volume annotations (create by e.g. downloading a volume project with multiple finished tasks)
 → should be merged
 → note that cell IDs are changed to be distinct
- click “view“ on a volume project with multiple finished tasks
 → should be merged, download should still give individual task data.zips
- click “copy to my account” on this compound volume project
 → should be merged, now download should also contain merged volume data
- test this also with fallback layer (also to check that ids >255 are handled correctly) (this is probably only possible via zip upload)

### Issues:
- fixes #3870 

------
